File: backend/app/schemas/ponto.py
from datetime import date, time
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel


class PontoCreate(BaseModel):
    usuario_id: int
    data: date
    entrada: time | None = None
    saida: time | None = None
    h_esperadas: Decimal = Decimal("8.0")


class PontoUpdate(BaseModel):
    entrada: time | None = None
    saida: time | None = None
    h_esperadas: Decimal | None = None


class PontoResponse(BaseModel):
    id: int
    usuario_id: int
    data: date
    entrada: time | None
    saida: time | None
    h_trabalhadas: Decimal | None
    h_esperadas: Decimal
    diferenca: Decimal | None
    situacao: str | None

    model_config = {"from_attributes": True}


class TotaisPonto(BaseModel):
    total_registros: int
    total_horas_trabalhadas: Decimal
    total_horas_esperadas: Decimal
    total_diferenca: Decimal
    faltas: int
    atrasos: int
    horas_extras: int
    normais: int

File: frontend/src/types/ponto.ts
export type PontoSituacao = 'normal' | 'atraso' | 'falta' | 'h_extra';

export interface RegistroPonto {
  id: number;
  usuario_id: number;
  data: string;
  entrada: string | null;
  saida: string | null;
  h_trabalhadas: number | null;
  h_esperadas: number;
  diferenca: number | null;
  situacao: PontoSituacao | null;
}

export interface PontoCreate {
  usuario_id: number;
  data: string;
  entrada?: string;
  saida?: string;
  h_esperadas?: number;
}

export interface PontoUpdate {
  entrada?: string;
  saida?: string;
  h_esperadas?: number;
}

export interface TotaisPonto {
  total_registros: number;
  total_horas_trabalhadas: number;
  total_horas_esperadas: number;
  total_diferenca: number;
  faltas: number;
  atrasos: number;
  horas_extras: number;
  normais: number;
}
